service-workers/service-worker/registration-updateviacache.https.html

```diff
@@ -0,0 +1,210 @@
+    [registerViaLinkHeader, 'via-link-header']
+  ];
+
+  // Test creating registrations.
+  for (const [registrationMethod, registrationMethodName] of registrationMethods) {
+    for (const updateViaCache of UPDATE_VIA_CACHE_VALUES) {
+      const testName = `register-${registrationMethodName}-updateViaCache-${updateViaCache}`;
+
+      promise_test(async t => {
+        await cleanup();
+
+        const opts = {scope: SCOPE};
+
+        if (updateViaCache) opts.updateViaCache = updateViaCache;
+
+        const reg = await registrationMethod(
+          `${SCRIPT_URL}?test=${testName}`,
+          opts
+        );
+
+        const sw = reg.installing || reg.waiting || reg.active;
+        await wait_for_state(t, sw, 'activated');
+        const values = await getScriptTimes(sw, testName);
+        await reg.update();
+
+        if (updateViaCache == 'all') {
+          assert_equals(reg.installing, null, "No new service worker");
+        }
+        else {
+          const newWorker = reg.installing;
+          assert_true(!!newWorker, "New worker installing");
+          const newValues = await getScriptTimes(newWorker, testName);
+
+          if (!updateViaCache || updateViaCache == 'imports') {
+            assert_not_equals(values.mainTime, newValues.mainTime, "Main script should have updated");
+            assert_equals(values.importTime, newValues.importTime, "Imported script should be the same");
+          }
+          else { // updateViaCache == 'none'
+            assert_not_equals(values.mainTime, newValues.mainTime, "Main script should have updated");
+            assert_not_equals(values.importTime, newValues.importTime, "Imported script should have updated");
+          }
+        }
+
+        await cleanup();
+      }, testName);
+    }
+  }
+
+  // Test changing the updateViaCache value of an existing registration.
+  for (const updateViaCache1 of UPDATE_VIA_CACHE_VALUES) {
+    for (const updateViaCache2 of UPDATE_VIA_CACHE_VALUES) {
+      const testName = `register-with-updateViaCache-${updateViaCache1}-then-${updateViaCache2}`;
+
+      promise_test(async t => {
+        await cleanup();
+
+        const fullScriptUrl = `${SCRIPT_URL}?test=${testName}`;
+        let opts = {scope: SCOPE};
+        if (updateViaCache1) opts.updateViaCache = updateViaCache1;
+
+        const reg = await navigator.serviceWorker.register(fullScriptUrl, opts);
+
+        const sw = reg.installing;
+        await wait_for_state(t, sw, 'activated');
+        const values = await getScriptTimes(sw, testName);
+
+        opts = {scope: SCOPE};
+        if (updateViaCache2) opts.updateViaCache = updateViaCache2;
+
+        await navigator.serviceWorker.register(fullScriptUrl, opts);
+
+        // If there's no change to the updateViaCache value, register should be a no-op.
+        // The default value should behave as 'imports'.
+        if ((updateViaCache1 || 'imports') == (updateViaCache2 || 'imports')) {
```

Comment:
Maybe we want to do this when (updateViaCache2 == 'all'), since it cannot pass the byte-check and thus there won't be a new service worker in that case.